### ISSUE TITLE
Could there be more options of saving webpages?

### ISSUE THREAD

**Issue by hyb897** [CitrusIce/ImportBookmarksToZotero#2]
I came across this issue when trying to save certain kind of webpages.(e.g. :https://www.mcmaster.com/shim-stock-sheets/material~301-stainless-steel/). For these kind of webpages, the save-to-zotero provides three options: save embeded metadata, save webpage w/ &w/o snapshot. It seems that the zotero use the first option by default, and when I check the saved entry in zotero，it only gives out the mainpage of the website(https://www.mcmaster.com/). But if I manually select the saving option to save the webpage, it works out normally, and I can save the entire url to zotero.

**Comment by CitrusIce** [comment]
The zotero actually have two options for saving web page, one is saving with translators and another is saving directly. The plugin first check if there has translators for saving, if not it saves webpage directly.

**Comment by CitrusIce** [comment]
you can try to remove the translator for that website

**Comment by hyb897** [comment]
> 

I think it's probably the embedded metadata translator. Will try to remove it to see the result.

**Comment by hyb897** [comment]
I tried to remove the translator so the zotero will just save webpage directly. Now it seems that when the program is opening the websites, the website will ask for credentials（ask you to log in) to browse the content, which I will not encounter if I open those websites myself.  Maybe the website is trying to protect itself from crawler？Is there any way to get around this， so I can add snapshots to my library？
![attached](https://user-images.githubusercontent.com/123704690/215303548-c6073050-5fab-495d-a976-81c971d8c0ed.png)
![attached2](https://user-images.githubusercontent.com/123704690/215303696-43d84d3a-37e9-480e-a056-cf6df5b8b6ac.png)

**Comment by CitrusIce** [comment]
I have added a new options for setting chrome userdata dir, set the userdata dir options to your chrome user data dir which have logged in that website and the script can use the creditional for the website, check the new branch https://github.com/CitrusIce/ImportBookmarksToZotero/tree/chromeuserdata_test. I have no time to test the new function and you need to test yourself

**Comment by hyb897** [comment]
Thank you for making the update to the code! I checked the code and it generally works well after minor correction at line 82:

`if self.userdata_dir: ` -> `if not self.userdata_dir:`

For the user data directory, I initially took it from chrome, but it seems that the version of chronium used by pyppeteer is not the latest version, so I have to create another user profile by directly open the chronium and logged in to the website. Then I copy the user data folder to my working folder and I could open the website without any problem. 

However, when compared to the entries I manually imported, I noticed that the titles of some of the entries imported through your code only display the name of the website, instead of the specific item name. (e.g.    McMaster-carr instead of Super-corrosion resistant 316 Stainless steel ......| McMaster-carr). This was solved by ‘page.bringToFront()’ before generating the 'all_script' for using zotero-connector).

As I don't have a technical background in computer science, I mostly rely on software as a tool to solve numerical problems. Your code, however, has provided me with a fantastic illustration of object-oriented programming, and I have gained a wealth of knowledge from it. I eagerly anticipate your future creations!
